mongo-statefulset/config_server/template.yaml

```diff
@@ -0,0 +1,165 @@
+kind: Template
+apiVersion: v1
+metadata:
+  name: mongodb-configserver-aws-on-openshift
+  annotations:
+    openshift.io/display-name: "AWS MongoDB Config Server"
+    description: "MongoDB Config Server on OpenShift, AWS Hosted"
+    iconClass: "icon-mongodb"
+    tags: "database,mongodb,nosql,config"
+parameters:
+
+  - name: MONGODB_CONFIGSVR_REPLICA_NAME
+    displayName: "ConfigServer Replica Set Name"
+    description: "The name of the configuration server replica set."
+    value: cs0
+    required: true
+
+  - name: MONGODB_ADMIN_PASSWORD
+    displayName: "MongoDB Admin Password"
+    description: "Password for the admin user."
+    generate: expression
+    from: "[a-zA-Z0-9]{16}"
+    required: true
+
+  - name: MONGODB_KEYFILE_VALUE
+    displayName: "Keyfile Content"
+    description: "The value of the MongoDB keyfile for inter-cluster communication. Must be the same throughout the cluster. See (https://docs.mongodb.com/manual/core/security-internal-authentication/#internal-auth-keyfile)."
+    generate: expression
+    from: "[a-zA-Z0-9]{255}"
+    required: true
+
+  - name: MONGODB_IMAGE
+    displayName: "MongoDB Docker Image"
+    description: "A reference to a supported MongoDB Docker image."
+    value: "bornemannjs/mongodb-34:0.71"
```

Comment:
PR(s) are still out and making their way in for the software collections project - so example uses my own build (for now)